pkg/controllers/internalmembercluster/v1beta1/member_controller_integration_test.go

```diff
@@ -2,220 +2,834 @@
+				}
+
+				return nil
+			}, eventuallyTimeout, eventuallyInterval).Should(Succeed(), "Failed to update the agent status")
+		})
+
+		It("can mark the cluster as left", func() {
+			// Add an offset of -1 second to avoid flakiness caused by approximation.
+			timeLeft = metav1.Time{Time: time.Now().Add(-time.Second)}
+
+			imc := &clusterv1beta1.InternalMemberCluster{}
+			objKey := types.NamespacedName{
+				Name:      member1Name,
+				Namespace: member1ReservedNSName,
+			}
+			Expect(hubClient.Get(ctx, objKey, imc)).Should(Succeed())
+
+			imc.Spec.State = clusterv1beta1.ClusterStateLeave
+			Expect(hubClient.Update(ctx, imc)).Should(Succeed())
+		})
+
+		It("should let the cluster go", func() {
+			Eventually(func() error {
+				imc := &clusterv1beta1.InternalMemberCluster{}
+				objKey := types.NamespacedName{
+					Name:      member1Name,
+					Namespace: member1ReservedNSName,
+				}
+				if err := hubClient.Get(ctx, objKey, imc); err != nil {
+					return fmt.Errorf("failed to get InternalMemberCluster: %w", err)
+				}
+
+				wantIMCStatus := clusterv1beta1.InternalMemberClusterStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:               string(clusterv1beta1.ConditionTypeClusterPropertyCollectionSucceeded),
+							Status:             metav1.ConditionTrue,
+							Reason:             ClusterPropertyCollectionSucceededReason,
+							Message:            ClusterPropertyCollectionSucceededMessage,
+							ObservedGeneration: imc.Generation,
+						},
+						{
+							Type:               string(clusterv1beta1.ConditionTypeClusterPropertyProviderStarted),
+							Status:             metav1.ConditionTrue,
+							Reason:             ClusterPropertyProviderStartedReason,
+							Message:            ClusterPropertyProviderStartedMessage,
+							ObservedGeneration: imc.Generation,
+						},
+						{
+							Type:               propertiesManuallyUpdatedConditionType,
+							Status:             metav1.ConditionFalse,
+							Reason:             propertiesManuallyUpdatedConditionReason2,
+							Message:            propertiesManuallyUpdatedConditionMsg2,
+							ObservedGeneration: imc.Generation,
+						},
+					},
+					Properties: map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+						propertyprovider.NodeCountProperty: {
+							Value: "2",
+						},
+					},
+					ResourceUsage: clusterv1beta1.ResourceUsage{
+						Capacity: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("15"),
+							corev1.ResourceMemory: resource.MustParse("30Gi"),
+						},
+						Allocatable: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("12"),
+							corev1.ResourceMemory: resource.MustParse("24Gi"),
+						},
+						Available: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("3"),
+							corev1.ResourceMemory: resource.MustParse("6Gi"),
+						},
+					},
```

Comment:
how should cx use those when the agent is not joined?

Reply:
Hi Ryan! At this moment the behavior is to leave the last observed data there; would you like me to update it so that the IMC controller would clear the information out when a cluster leaves?